public/static-checkout/js/scripts.js

```diff
@@ -0,0 +1,309 @@
+} else if (window.location.hostname === 'payment.stage.edx.org') {
+  ecommerceBaseUrl = 'https://ecommerce.stage.edx.org';
+} else if (window.location.hostname === 'localhost') {
+  ecommerceBaseUrl = 'http://localhost:18130';
+}
+
+function getBasketId() {
+  var queryString = window.location.search;
+  var urlParams = new URLSearchParams(queryString);
+  // Emma: sql or other injection?
+  var basketId = urlParams.get('basketId');
+  // Emma: what if the basketId is not in the param?
+  return basketId;
+}
+
+function redirectToMFE(couponCode) {
+  var queryString = window.location.search;
+  var urlParams = new URLSearchParams(queryString);
+  var sku = urlParams.get('sku');
+  var newUrl = ecommerceBaseUrl + '/basket/add/?sku=' + sku;
+  if (couponCode) {
+    newUrl = newUrl.concat('&coupon=', couponCode);
+  }
+  window.location.href = newUrl;
+}
+
+userButton.addEventListener('click', function toggleUserDropdown() {
+  document.querySelector('.dropdown-menu').classList.toggle('show');
+}, false);
+
+window.onclick = function closeMenu(event) {
+  if (!event.target.matches('.user-button')) {
+    var dropdownMenu = document.getElementsByClassName('dropdown-menu');
+    var i;
+    for (i = 0; i < dropdownMenu.length; i += 1) {
+      var open = dropdownMenu[i];
+      if (open.classList.contains('show')) {
+        open.classList.remove('show');
+      }
+    }
+  }
+};
+
+function renderStates(country) {
+  var states = Object.values(countryStates[country]);
+  for (var i = 0; i < states.length; i += 1) {
+    var state = document.createElement('option');
+    state.innerHTML = states[i];
+    stateSelect.appendChild(state);
+  }
+}
+
+function renderCountryState() {
+  countrySelect.onchange = function getCountryStates() {
+    var stateSelectHtml = document.createElement('div');
+    stateSelectHtml.innerHTML = '<select id="state" name="state"><option value="" selected="selected">Choose state/province</option></select>';
+    stateSelect.length = 1;
+    if (!(countrySelect.value === 'CA' || countrySelect.value === 'US')) {
+      stateSelectHtml = document.createElement('div');
+      stateSelectHtml.innerHTML = '<input id="state" type="text" name="state">';
+      stateSelect.parentNode.replaceChild(stateSelectHtml, stateSelect);
+      stateSelect = document.getElementById('state');
+    } else {
+      stateSelect.parentNode.replaceChild(stateSelectHtml, stateSelect);
+      stateSelect = document.getElementById('state');
+      if (countrySelect.value === 'CA') {
+        renderStates('CA');
+      } else if (countrySelect.value === 'US') {
+        renderStates('US');
+      }
+    }
+  };
+}
+
+document.addEventListener('DOMContentLoaded', function handleCountrySelection() {
+  renderCountryState();
+});
+
+couponButton.addEventListener('click', function couponRedirectToMicrofrontend(event) {
+  event.preventDefault();
+  var couponCode = document.getElementById('couponcode').value;
+  redirectToMFE(couponCode);
```

Comment:
Is redirecting to the MFE for carts that use coupons going to skew the results?  Seems like all those carts would have lower value on average.